dspace-spring-rest/src/main/java/org/dspace/app/rest/OpenSearchController.java

```diff
@@ -0,0 +1,245 @@
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import org.w3c.dom.Document;
+
+/**
+ * This class provides a controller for OpenSearch support.
+ * It creates a namespace /opensearch in the DSpace REST webapp.
+ *
+ * @author Oliver Goldschmidt (o.goldschmidt at tuhh.de)
+ */
+@Controller
+@RequestMapping("/opensearch")
+public class OpenSearchController implements ErrorController {
+
+    private static final Logger log = Logger.getLogger(ScopeResolver.class);
+    private static final String errorpath = "/error";
+    private List<String> searchIndices = null;
+
+    private CommunityService communityService;
+    private CollectionService collectionService;
+    private AuthorizeService authorizeService;
+    private OpenSearchService openSearchService;
+
+    private Context context;
+
+    /**
+     * This method provides the OpenSearch query on the path /search
+     * It will pass the result as a OpenSearchDocument directly to the client
+     */
+    @GetMapping("/search")
+    public void search(HttpServletRequest request,
+                         HttpServletResponse response,
+                         @RequestParam(name = "query", required = false) String query,
+                         @RequestParam(name = "start", required = false) Integer start,
+                         @RequestParam(name = "rpp", required = false) Integer count,
+                         @RequestParam(name = "format", required = false) String format,
+                         Model model) throws IOException, ServletException {
+        context = ContextUtil.obtainContext(request);
+        if (start == null) {
+            start = 0;
+        }
+        if (count == null) {
+            count = -1;
+        }
+        if (openSearchService == null) {
+            openSearchService = UtilServiceFactory.getInstance().getOpenSearchService();
+        }
+        if (openSearchService.isEnabled()) {
+            init();
+            // get enough request parameters to decide on action to take
+            if (format == null || "".equals(format)) {
+                // default to atom
+                format = "atom";
+            }
+
+            log.debug("Searching for " + query + " in format " + format);
+
+            // do some sanity checking
+            // TODO: make that work correctly
+            if (!openSearchService.getFormats().contains(format)) {
+                String err = "Format " + format + " is not supported.";
+                response.setContentType("text/html");
+                response.setContentLength(err.length());
+                response.getWriter().write(err);
+                //response.sendError(HttpServletResponse.SC_BAD_REQUEST);
+                //return "error";
```

Comment:
Are these commented out lines still needed?

Reply:
No, they are not. I have removed them.